tests/utils/test_concat_mla.py

```
@@ -187,5 +312,65 @@ def test_concat_mla_k_dtype_mismatch_raises():
+    )
+    assert torch.equal(
+        k_rope.contiguous().view(torch.uint8),
+        k_rope_snapshot.contiguous().view(torch.uint8),
+    )
+
+
+def test_cake_concat_mla_k_rejects_non_sm103a(monkeypatch):
+    _require_cake_concat_mla_k()
+    k, k_nope, k_rope = _make_cake_tensors(1, torch.bfloat16, "contiguous", False)
+    monkeypatch.setattr(torch.cuda, "get_device_capability", lambda _device: (10, 0))
+    with pytest.raises(RuntimeError, match="exact compute capability 10.3"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)